src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+            final int locusEnd = newEventLocus.getLocusEnd();
+
+            // clean out old stuff that can't possibly be interesting anymore
+            final Iterator<Map.Entry<EventLocus,GATKRead>> itr = locMap.entrySet().iterator();
+            final int staleEnd = locusStart - MAX_LOCUS_DIST;
+            while ( itr.hasNext() )
+            {
+                final EventLocus eventLocus = itr.next().getKey();
+                if ( eventLocus.getLocusStart() >= locusStart )
+                    break;
+                if ( eventLocus.getLocusEnd() <= staleEnd )
+                    itr.remove();
+            }
+
+            locMap.put(newEventLocus.makeUnique(++eventIdCounter),read);
+
+            // see if there are a sufficient number of overlapping putative events to call a cluster
+            final List<Map.Entry<EventLocus,GATKRead>> overlappers = new LinkedList<>();
+            for ( Map.Entry<EventLocus,GATKRead> entry : locMap.entrySet() )
+            {
+                final EventLocus eventLocus = entry.getKey();
+                if ( eventLocus.getLocusStart() >= locusEnd )
+                    break;
+                if ( eventLocus.getLocusEnd() > locusStart )
+                    overlappers.add(entry);
+            }
+            if ( overlappers.size() >= MIN_EVIDENCE )
+            {
+                final List<GATKRead> list = new LinkedList<>();
+                for ( Map.Entry<EventLocus,GATKRead> entry : overlappers )
+                {
+                    if (entry.getValue() != null)
+                    {
+                        list.add(entry.getValue());
+                        entry.setValue(null);
+                    }
+                }
+                return list.iterator();
+            }
+            return Collections.emptyIterator();
+        }
+    }
+
+    // Class that acts as a mapper from a stream of reads to a stream of <Kmer,BreakpointId> pairs.
+    // Sitting atop the BreakpointClusterer, it groups funky reads that are mapped near each other and assigns that
+    // group a breakpoint id, and transforms each funky read into kmers which are emitted along with the breakpoint id.
+    private static final class ClusterKmerizer implements Iterator<Tuple2<Kmer,Long>>, Iterable<Tuple2<Kmer,Long>>
+    {
+        private final Iterator<GATKRead> inputIterator;
+        private final Set<Kmer> killSet;
+        private Iterator<Tuple2<Kmer,Long>> outputIterator = Collections.emptyIterator();
+        private String currentContig = null;
+        private int lastLocus = 0;
+        private Long breakpointId;
+
+        private static final int CLUSTER_BREAK_DISTANCE = 500;
+        //TODO: Should depend on fragment size
+
+        public ClusterKmerizer( Integer partIdx, Iterator<GATKRead> someInputIterator, Set<Kmer> someKillSet )
+        {
+            inputIterator = someInputIterator;
+            killSet = someKillSet;
+            breakpointId = partIdx.longValue() << 32;
```

Comment:
Add a note explaining this and preferably extract the 32 as a named constant.  Bitwise operations are unusual in gatk and should always be explained.